### PR TITLE
[DO NOT MERGE] Speedup logging

### PR DESCRIPTION
**Pull request by Ben-Epstein**
# DO NOT MERGE

* [ ] I have added tests to `tests` to cover my changes.
* [ ] I have updated `docs/`, if necessary.
* [ ] I have updated the `README.md`, if necessary.

***What existing issue does this pull request close?***

Put `closes #issue-number` in this pull request's description to auto-close the issue that this fixes.

***How are these changes tested?***

This pull request includes automated tests for the code it touches and those tests are described below. If no tests are included, reasons why must be provided below.

These changes are tested with [...]

***Demonstration***

Demonstrate your contribution.

For example, what are the exact commands you ran and their output, related screenshots, screen-recordings, test runs, anything that can showcase.

***Provide additional context.***

Provide as much relevant context as you like.

### CODE REVIEW

File: dataquality/__init__.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 __version__ = "v0.8.14"
 
 import os
+
+os.environ["GALILEO_MULTI_PROC"] = "True"
```
Comment (anthonycorletti) [inline dataquality/__init__.py:7]: simpler to case on bool types in an env as 0 or 1 🤷

File: dataquality/utils/log_manager.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    For TC, MLTC, and IC, we use a ThreadPoolExecutor, because the majority of the work
+    is in I/O, and these tasks require write access to global variables (logger_config
+    vars like `observed_num_labels` and `observed_ids`
+
+    For NER, we use a ProcessPoolExecutor because the majority of the work is CPU bound,
+    in `process_sample` (see TextNERModelLogger), not I/O. It does NOT need any global
+    variable write access, so it's safe to be using a ProcessPoolExecutor
+    """
+
+    MAX_LOGGERS = 2
```
Comment (anthonycorletti) [inline dataquality/utils/log_manager.py:27]: should make the processpool and threadpool executors configurable to some non zero number of workers.

File: dataquality/utils/log_manager.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @staticmethod
+    def add_logger(target: Callable, task_type: TaskType) -> None:
+        """
+        Start a new function in a thread and store that in the global list of threads
+
+        :param target: The callable
+        :param args: The arguments to the function
+        :return: None
+        """
+        multi_proc = environ.get("GALILEO_MULTI_PROC", 1) in ("True", "TRUE", "true", 1)
```
Comment (anthonycorletti) [inline dataquality/utils/log_manager.py:41]: `TrUe`?

**Review by anthonycorletti — APPROVED** [review]
looks great! comments inline.

File: dataquality/utils/log_manager.py (hunk 3 above)
Comment (Ben-Epstein) [reply dataquality/utils/log_manager.py:41]: 😂 those people don't deserve quality data

File: dataquality/__init__.py (hunk 1 above)
Comment (Ben-Epstein) [reply dataquality/__init__.py:7]: yea :( it's a weird thing that python doesn't have a builtin for managing truthyness in env vars

File: dataquality/utils/log_manager.py (hunk 3 above)
Comment (anthonycorletti) [reply dataquality/utils/log_manager.py:41]: Yeah!

Comment (Ben-Epstein) [reply dataquality/utils/log_manager.py:41]: i updated it to be smarter, even those _those people_

**Review by franz101 — COMMENTED** [review]
From my first overview this looks good. I have a couple of questions regarding processes.
Are we spawning or forking? Is it common to use a named context for the processes?

File: dataquality/utils/log_manager.py (hunk 3 above)
Comment (anthonycorletti) [reply dataquality/utils/log_manager.py]: you're too gracious @Ben-Epstein

**Pull request closed without merging**